storage/innobase/mtr/mtr0mtr.cc

```diff
@@ -1217,19 +1217,35 @@ inline void log_t::resize_write(lsn_t lsn, const byte *end, size_t len,
     if (!resize_flush_buf)
     {
       ut_ad(is_pmem());
```

Comment:
I have a general question/suggestion about testing the is_pmem() T/F path on linux. Since the paths are significantly different, can we have a set of testcases (redo and recovery related) to be forced to run 2 times following each path ? Currently on linux, I see that is_pmem() path is followed when I use RAM disk (./mtr --mem). Probably CI also runs same way.

Reply:
We could add a debug build specific parameter for enabling or disabling the special handling of  `/dev/shm` . That would allow both code paths to be covered easily. Currently, I think that on any qualifying 64-bit Linux platform, https://buildbot.mariadb.org and https://buildbot.mariadb.net would run with memory-mapped files on `/dev/shm`. That is, this code path (and not the `pread` and `pwrite` based one) is what is being tested most.

Reply:
To test the non-memory-mapped log writes on RAM disk, the following is possible already, and used by me:
```sh
mkdir /run/user/$UID/var
rm -fr mysql-test/var
ln -s /run/user/$UID/var mysql-test/var
```
In my current Debian Sid system, I could also symlink `mysql-test/var` to a subdirectory of `/tmp`, which would be on RAM disk nowadays.

There is no PMEM support on at least one 64-bit Linux architecture on https://buildbot.mariadb.org: IBM System Z (s390x). We also have a 32-bit Linux target (x86) where we do not support PMEM. We also have FreeBSD, IBM AIX, Apple macOS without PMEM support. I think that the regular file I/O path is adequately covered on our CI systems.

If a production user stores the write-ahead log in `/dev/shm`, they should not care which impelmentation is being used. I wanted to implement `/dev/shm` as a special case so that we will get adequate test coverage of the PMEM code path, and so that there is a convenient way to test the relative performance of memory-mapped and `pwrite` based log writes.

Reply:
Ok. For the record let me cut-paste the messages from slack discussion that would help us identifying if mapped redo memory is used.

```
2024-09-04 14:22:12 0 [Note] InnoDB: File system buffers for log disabled (block size=512 bytes)
2024-09-04 14:22:12 0 [Note] InnoDB: End of log at LSN=58172

2024-09-04 14:22:56 0 [Note] InnoDB: log sequence number 56287 (memory-mapped); transaction id 16
```